### PR TITLE
feat: daemon call leaveHost when exit

### PR DESCRIPTION
**Pull request by bigerous**
Signed-off-by: bigerous <cuidajun.cdj@alibaba-inc.com>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: client/daemon/daemon.go [hunk 1]
```diff
@@ -765,6 +765,17 @@ func (cd *clientDaemon) Serve() error {
 func (cd *clientDaemon) Stop() {
 	cd.once.Do(func() {
 		close(cd.done)
+		if cd.schedulerClient != nil {
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+			defer cancel()
+			_ = cd.schedulerClient.LeaveHost(ctx, &schedulerv1.LeaveHostRequest{Id: cd.schedPeerHost.Id})
```
Comment (gaius-qi) [inline client/daemon/daemon.go:771]: Record error log.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: pkg/rpc/scheduler/client/client.go [hunk 2]
```diff
@@ -191,7 +191,7 @@ func (c *client) LeaveHost(ctx context.Context, req *schedulerv1.LeaveHostReques
 		wg.Add(1)
 		go func(addr string) {
 			defer wg.Done()
-			if err := c.leaveHost(trace.ContextWithSpan(context.Background(), trace.SpanFromContext(ctx)), addr, req, opts...); err != nil {
+			if err := c.leaveHost(trace.ContextWithSpan(ctx, trace.SpanFromContext(ctx)), addr, req, opts...); err != nil {
```
Comment (gaius-qi) [inline pkg/rpc/scheduler/client/client.go:194]:
```go
c.leaveHost(ctx, addr, req, opts...)
```

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by jim3ma — APPROVED** [review]
LGTM

**Pull request merged**